service/history/shard/controller_test.go

```diff
@@ -869,3 +1006,17 @@ func (m updateShardRequestMatcher) Matches(x interface{}) bool {
 func (m updateShardRequestMatcher) String() string {
 	return fmt.Sprintf("%+v", (persistence.UpdateShardRequest)(m))
 }
+
+func (s *controllerSuite) readMetricsCounter(name string, nonSystemTags ...metrics.Tag) float64 {
+	expectedSystemTags := []metrics.Tag{
+		metrics.StringTag("otel_scope_name", "temporal"),
+		metrics.StringTag("otel_scope_version", ""),
+	}
```

Comment:
we should really put these in some central place instead of spread across tons of tests... ah well, not in this PR

Reply:
Yeah, agreed. (I added the metricstest package to the repo, so I would like for it to be easier to use in tests.)